context/llpcCompiler.h

```diff
@@ -263,6 +263,42 @@ class Compiler: public ICompiler
     LLPC_DISALLOW_DEFAULT_CTOR(Compiler);
     LLPC_DISALLOW_COPY_AND_ASSIGN(Compiler);
 
+    // Class of object to manage checking and updating shader caches for graphics pipeline.
+    class GraphicsShaderCacheChecker
+    {
+    public:
+        GraphicsShaderCacheChecker(Compiler* pCompiler, Context* pContext) :
+            m_pCompiler(pCompiler), m_pContext(pContext)
+        {}
+
+        // Check shader caches, returning mask of which shader stages we want to keep in this compile.
+        uint32_t Check(const llvm::Module*                     pModule,
+                       uint32_t                                stageMask,
+                       llvm::ArrayRef<llvm::ArrayRef<uint8_t>> stageHashes);
+
+        // Get cache results.
+        ShaderEntryState GetNonFragmentCacheEntryState() { return m_nonFragmentCacheEntryState; }
+        ShaderEntryState GetFragmentCacheEntryState() { return m_fragmentCacheEntryState; }
+
+        // Update shader caches with results of compile, and merge ELF outputs if necessary.
+        void UpdateAndMerge(Result result, ElfPackage* pPipelineElf);
+
+    private:
+        static constexpr uint32_t ShaderCacheCount = 2;
+        Compiler* m_pCompiler;
+        Context*  m_pContext;
+
+        ShaderEntryState m_nonFragmentCacheEntryState = ShaderEntryState::New;
+        ShaderCache* m_pNonFragmentShaderCache[ShaderCacheCount] = {};
```

Comment:
It conflict with my change, now, compiler doesn't responsible for application pipeline cache.
so we only need check our internal cache, i.e. ShaderCacheCount always is 1.

Reply:
OK, I have rebased it on your #288, as I guess that will be merged soon.